devito/dimension.py

```diff
@@ -372,14 +372,14 @@ def _arg_defaults(self, **kwargs):
         A :class:`ConditionalDimension` provides no arguments, so this
         method returns an empty dict.
         """
-        return {}
+        return {self.size_name: kwargs.get('size')}
```

Comment:
I have no idea why you have to do this

Reply:
Because you need to input the size of the subdim